<launch>
    <!-- ros2 bag play _2022-04-07-14-14-35_robotarium/_2022-04-07-14-14-35_robotarium_ros2.db3 
    -\-rate 1.0 
    -\-clock 
    -\-start-paused 
    -\-topics /stereo_camera/left/image_raw /odometry/filtered 
    -\-remap /odometry/filtered:=/robotarium/odom -->

    <!-- Auxiliar node to convert from sensor_msgs/msg/Image to sensor_msgs/msg/CompressedImage  -->
    
    <node
        pkg="image_transport"
        exec="republish"
        name="image_republish"
        output="screen">
        <param name="input_transport" value="raw"/>
        <param name="output_transport" value="compressed"/>
        <param name="in_video_mode" value="raw"/>
        <param name="out_video_mode" value="compressed"/>
        <remap from="in" to="/stereo_camera/left/image_raw"/>
        <remap from="out" to="/robotarium/camera/image"/>
    </node>

    <node
        pkg="ratslam"
        exec="ratslam_lv"
        name="ratslam_view_template"
        output="screen"
        args="--ros-args --remap _image_transport:=compressed">
        <param name="use_sim_time" value="true" />
        <param name="topic_root" value="robotarium" />
        <param name="media_path" value="$(find-pkg-share ratslam)/media" />
        <param name="image_file" value="irat_sm.tga" />
        <param from="$(find-pkg-share ratslam)/config/config_robotarium.yaml" />
    </node>

    <node
        pkg="ratslam"
        exec="ratslam_pc"
        name="ratslam_pose_cells"
        output="screen"
        args="--ros-args --remap _image_transport:=compressed">
        <param name="use_sim_time" value="true" />
        <param name="topic_root" value="robotarium" />
        <param name="media_path" value="$(find-pkg-share ratslam)/media" />
        <param name="image_file" value="irat_sm.tga" />
        <param from="$(find-pkg-share ratslam)/config/config_robotarium.yaml" />
    </node>

    <node
        pkg="ratslam"
        exec="ratslam_em"
        name="ratslam_experience_map"
        output="screen"
        args="--ros-args --remap _image_transport:=compressed">
        <param name="use_sim_time" value="true" />
        <param name="topic_root" value="robotarium" />
        <param name="media_path" value="$(find-pkg-share ratslam)/media" />
        <param name="image_file" value="irat_sm.tga" />
        <param from="$(find-pkg-share ratslam)/config/config_robotarium.yaml" />
    </node>
</launch>

